spring-cloud-stream-app-starters-docs/src/main/asciidoc/overview.adoc

```diff
@@ -251,4 +251,230 @@ public class DemoApplication {
 
 * Build and install the application to desired maven repository
 * The patched copy of `jdbc-sink` application now includes `mysql` driver in it
-* This application can be run as standalone _uberjars_
\ No newline at end of file
+* This application can be run as standalone _uberjars_
+
+=== Developing custom applications and generating artifacts using the existing tooling
+
+In this section, we will explain how to develop a custom source/sink/processor application and then generate
+maven and docker artifacts for it with the necessary middleware bindings using the existing tooling provided by the
+spring cloud stream app starter infrastructure. For explanation purposes, we will assume that we are creating a new
+source application for a technology named foobar.
+
+* Create a new top level module named spring-cloud-starter-stream-source-foobar
+
+Please follow the instructions above for designing a proper Spring Cloud Stream Source. You may also look into the existing
+starters for how to structure a new one. Ensure that you name the main `@Configuration` class of your starter as
+`FoobarSourceConfiguration` as this is the default convention used by the app generation later. Please also make sure
+that the package for this `@Configuration` is `org.springfamework.cloud.stream.app.foobar.source`. If you have a different
+class/package name, see below for how to override that in the app generator. The functionality name for which you create
+a starter can be a hyphenated stream of strings such as in `scriptable-transform` which is a processor type in the
+module `spring-cloud-starter-stream-processor-scriptable-transform`.
+
+The starters in `spring-cloud-stream-app-starters` are slightly different from the other starters in spring-boot and
+spring-cloud in that here we don't provide a way to auto configure any configuration through spring factories mechanism.
+Rather, we delegate this responsibility to the maven plugin that is generating the binder based apps. Therefore, you don't
+have to provide a spring.factories file that lists all your configuration classes.
+
+* Add the new foobar source module to the root pom of the repository
+
+* At the root of the repository build and install the new module into your local maven cache:
+
+`./mvnw clean install -pl :spring-cloud-starter-stream-source-foobar`
+
+* You need to add the new starter dependency to the `spring-cloud-stream-app-dependencies` bill of material (BOM) in the
+dependecy management section. For example,
+
+[source,xml]
+----
+<dependencyManagement>
+...
+...
+    <dependency>
+        <org.springframework.cloud.stream.app>
```

Comment:
Missing the element tag names such as `artifactId`, `groupId` and `version`. Is that intentional?